### DISCUSSION TITLE
How can I deal with not approved pull requests with feature dependencies?

### DISCUSSION THREAD

**Discussion by zaldivards** [community/community#22810] [New to GitHub]
<p>Suppose i have a <strong>feature A</strong> and make a push to the remote server and then a PR with <strong>develop</strong> branch, then someone assign to me <strong>feature B</strong> which it needs some of the code on <strong>feature A</strong> , but <strong>feature A</strong> is not approved yet.</p>
<p>How can i manage this? Should i create a new local branch upon feature <strong>A</strong> instead of <strong>develop</strong> ? Or what would be the best practice?</p>
<p>By the way, I’m using <strong>gitflow workflow</strong></p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="Zaldivar97" data-post="1" data-topic="135148">
<div class="title">
<div class="quote-controls"></div>
 Zaldivar97:</div>
<blockquote>
<p>Should i create a new local branch upon feature <strong>A</strong> instead of <strong>develop</strong> ?</p>
</blockquote>
</aside>
<p>That would be common practice, yes. Two things to be aware of:</p>
<ol>
<li>If branch A is rebased (including rebase-merge), or squash-merged into develop your branch B will contain commits originating from branch A that probably shouldn’t become part of history. You’ll have to clean that up before merging branch B into develop.</li>
<li>Similarly, if there is still development (as in, new commits) on branch A you might have to merge them into branch B, or rebase B, to make sure you have the latest state. Technically you could keep developing B and look at the differences when it’s time to merge, but usually it’s easier to handle them early.</li>
</ol>
